Lib/Protocols/IdFTP.pas

```diff
@@ -2742,18 +2771,31 @@ procedure TIdFTP.IssueFEAT;
       Break;
     end;
   end;
+  if FClientInfo.ClientName <> '' then begin
+    if IsExtSupported('CSID') then begin {do not localize}
 
+      SendCmd('CSID ' + FClientInfo.GetCSIDOutput);  {do not localize}
+      if Self.LastCmdResult.NumericCode = 200 then begin
+        LBuf := TrimRight(Self.LastCmdResult.Text.Text);
+        repeat
+           FServerInfo.ServerInfo.Add(TrimLeft(Fetch(LBuf,';')));
+        until LBuf = '';
+      end;
+    end
+    else
+    begin
   // send the CLNT command before sending the OPTS UTF8 command.
   // some servers need this in order to work around a bug in
   // Microsoft Internet Explorer's UTF-8 handling
-  if IsExtSupported('CLNT') then begin {do not localize}
-    LClnt := FClientInfo.ClntOutput;
-    if LClnt = '' then begin
-      LClnt := gsIdProductName + ' ' + gsIdVersion;
+      if IsExtSupported('CLNT') then begin {do not localize}
+        LClnt := FClientInfo.ClntOutput;
+        if LClnt = '' then begin
+          LClnt := gsIdProductName + ' ' + gsIdVersion;

```

Comment:
Since you are now checking `FClientInfo.ClientName` beforehand, `FClientInfo.ClntOutput` can never return a blank string here anymore, so using Indy's product/version here is now dead code.

On the other hand, the original code would always send `CLNT` if supported by the server, and if `FClientInfo.ClientName` is empty then Indy's default product/version is sent.  That logic should probably be preserved.  In which case, I would suggest also making `CSID` use Indy's product/version as well if `FClientInfo.ClientName` is empty.